internal/provisioner/certmanager.go

```diff
@@ -0,0 +1,170 @@
+		desiredVersion: desiredVersion,
+	}, nil
+
+}
+
+func (n *certManager) updateVersion(h *helmDeployment) error {
+	actualVersion, err := h.Version()
+	if err != nil {
+		return err
+	}
+
+	n.actualVersion = actualVersion
+	return nil
+}
+
+func (n *certManager) Create() error {
+	h := n.NewHelmDeployment()
+	err := h.Create()
+	if err != nil {
+		return err
+	}
+
+	err = n.updateVersion(h)
+	return err
+}
+
+func (n *certManager) Upgrade() error {
+	h := n.NewHelmDeployment()
+	err := h.Update()
+	if err != nil {
+		return err
+	}
+
+	err = n.updateVersion(h)
+	return err
+}
+
+func (n *certManager) DesiredVersion() string {
+	return n.desiredVersion
+}
+
+func (n *certManager) ActualVersion() string {
+	return strings.TrimPrefix(n.actualVersion, "cert-manager-")
+}
+
+func (n *certManager) Destroy() error {
+	return nil
+}
+
+func (n *certManager) NewHelmDeployment() *helmDeployment {
+	return &helmDeployment{
+		chartDeploymentName: "cert-manager",
+		chartName:           "jetstack/cert-manager",
+		namespace:           "cert-manager",
+		setArgument:         "",
+		valuesPath:          "helm-charts/cert-manager_values.yaml",
+		kopsProvisioner:     n.provisioner,
+		kops:                n.kops,
+		logger:              n.logger,
+		desiredVersion:      n.desiredVersion,
+	}
+}
+
+func (n *certManager) Name() string {
+	return model.CertManagerCanonicalName
+}
+
+func deployCertManagerCRDS(kops *kops.Cmd, logger log.FieldLogger) error {
+	files := []k8s.ManifestFile{
+		{
+			Path:            "certmanager-manifests/crds.yaml",
+			DeployNamespace: "cert-manager",
+		},
+	}
+	k8sClient, err := k8s.New(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return err
+	}
+	logger.Infof("Deploying cert manager crds")
+	err = k8sClient.CreateFromFiles(files)
+	if err != nil {
+		return err
+	}
+	logger.Infof("Successfully deployed cert manager crds")
+	return nil
+}
+
+func deployClusterIssuer(kops *kops.Cmd, logger log.FieldLogger) error {
+	k8sClient, err := k8s.New(kops.GetKubeConfigPath(), logger)
+	if err != nil {
+		return err
+	}
+	
+	wait := 300
+	pods, err := k8sClient.GetPodsFromDeployment("cert-manager", "cert-manager-webhook")
+	if err != nil {
+		return err
+	}
+	if len(pods.Items) == 0 {
+		return fmt.Errorf("no pods found from cert-manager-webhook deployment")
+	}
+
+	for _, pod := range pods.Items {
+		logger.Infof("Waiting up to %d seconds for cert-manager-webhook pod %q to start...", wait, pod.GetName())
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+		defer cancel()
+		pod, err := k8sClient.WaitForPodRunning(ctx, "cert-manager", pod.GetName())
+		if err != nil {
+			return err
+		}
+		logger.Infof("Successfully deployed cert-manager-webhook pod %q", pod.Name)
+	}
+
+	//After the webhook is up and running it needs ~2sec before being able to accept connections and being able to deploy the cluster issuer.
+	time.Sleep(5 * time.Second)
```

Comment:
Can we poll the endpoint until it comes up, or something like that, instead of just a sleep? 0/5 on this, just a thought.

Reply:
I don't like it either but the thing is that as soon as the pod is app it takes it like a second to prepare the hook. Its not sth changing in every run, its an application thing.

Reply:
i will check this check is being made in the cluster-api, I think I saw something like that

Reply:
you can do something like
```
$ kubectl wait --for condition=Available --timeout 300s apiservice v1alpha2.cert-manager.io
```
this will return when the condition met, and error when timeout


just need to check what is the service you want to check

Reply:
thats great thanks, I will have a look.

Reply:
Pushed a fix for that